lib/miq_automation_engine/models/miq_ae_domain.rb

```diff
@@ -200,4 +220,21 @@ def self.import_options(git_repo, options)
   end
 
   private_class_method :import_options
+
+  def self.validate_refs(repo, options)
+    match = nil
+    case options['ref_type'].downcase
```

Comment:
This whole format feels awkward.  Instead of a hash like

`{"ref_type" => "branch", "ref" => "foo"}` or `{"ref_type" => "tag", "ref" => "bar"}`

what if instead it was structured like

`{"branch" => "foo"}` or `{"tag" => "bar"}`

Then this code and the code above doesn't need to do weird stuff with ref_type.  What do you think?